plugins/tech-insights-backend/src/service/router.ts

```diff
@@ -103,16 +103,30 @@ export async function createRouter<
         checks,
         entities,
       }: { checks: string[]; entities: CompoundEntityRef[] } = req.body;
+      let error = undefined;
       const tasks = entities.map(async entity => {
         const entityTriplet =
           typeof entity === 'string' ? entity : stringifyEntityRef(entity);
-        const results = await factChecker.runChecks(entityTriplet, checks);
-        return {
-          entity: entityTriplet,
-          results,
-        };
+        try {
+          const results = await factChecker.runChecks(entityTriplet, checks);
+          return {
+            entity: entityTriplet,
+            results,
+          };
+        } catch (e: any) {
+          const errorMessage = `Failed to run check for entity ${entityTriplet} due to error: ${e.message}`;
+          logger.error(errorMessage);
+          error = errorMessage;
```

Comment:
It is probably better to gather error messages for all entities  that are failing. This would return information only about the last failed entity

Reply:
I'll also mention that the `@backstage/errors` package contains a [`serializeError`](https://github.com/backstage/backstage/blob/bb1122ba22f4e0a8c38413f154aadb3b74f63174/packages/errors/src/serialization/error.ts#L47) function that emits an error as a standard-shaped object that's ready to be sent as JSON over the wire. This is the format that for example our standard error handling middleware for backends emits them, and the frontend also has special tooling for that format